airbyte-integrations/connectors/destination-s3-v2/src/main/kotlin/S3V2Checker.kt

```diff
@@ -4,12 +4,14 @@
 
 package io.airbyte.integrations.destination.s3_v2
 
+import io.airbyte.cdk.load.command.s3.S3Client
 import io.airbyte.cdk.load.check.DestinationChecker
 import jakarta.inject.Singleton
+import kotlinx.coroutines.runBlocking
 
 @Singleton
-class S3V2Checker : DestinationChecker<S3V2Configuration> {
+class S3V2Checker(private val s3Client: S3Client) : DestinationChecker<S3V2Configuration> {
     override fun check(config: S3V2Configuration) {
-        // TODO: validate that the configuration works
+        runBlocking { s3Client.list("") }
```

Comment:
future pr: ideally, check is somehow kicking off a mini-sync, then deleting its output, rather than doing this sort of oneoff capability check

Reply:
(part of me wants to call `AirbyteDestinationRunner.run("write")` from inside `check`, which is probably a crime somewhere)